airstack-modules/modules/airstack/social/social.ts

```diff
@@ -1,998 +1,1226 @@
+     * @param tokenId erc721 token id,empty string if you've to remove defaultProfile
+     * @param tokenAddress erc721 token address
+     * @param socialUserId social user id (eg: farcasterId)
+     */
+    export function trackSocialUserDefaultProfileChange(
+        block: ethereum.Block,
+        transactionHash: string,
+        logOrCallIndex: BigInt,
+        from: string,
+        to: string,
+        tokenId: string,
+        tokenAddress: string,
+        socialUserId: string
+    ): void {
+        const chainId = getChainId()
+        const airBlock = getOrCreateAirBlock(
+            chainId,
+            block.number,
+            block.hash.toHexString(),
+            block.timestamp
+        )
+        airBlock.save()
+        let newDefaultProfile: AirSocialProfile | null = null
+        if (tokenId != "") {
+            newDefaultProfile = getAirSocialProfile(chainId, tokenAddress, tokenId)
+            if (newDefaultProfile == null) {
+                throw new Error("air social profile not found")
+            }
+        }
+        const airSocialUser = getAirSocialUser(chainId, socialUserId)
+        if (airSocialUser == null) {
+            throw new Error("air social user not found")
+        }
+        createAirSocialUserDefaultProfileChangeTransaction(
+            chainId,
+            airBlock,
+            transactionHash,
+            logOrCallIndex,
+            from,
+            to,
+            airSocialUser,
+            newDefaultProfile
+        )
+        airSocialUser.defaultProfile = newDefaultProfile != null ? newDefaultProfile.id : null
+        airSocialUser.save()
     }
-    // create air social user
-    const airSocialUser = createAirSocialUser(chainId, airBlock, socialUserId, to, airSocialUserExtraIds);
-    // create air social profile
-    const airSocialProfile = createAirSocialProfile(airBlock, chainId, airSocialUser.id, profileName, tokenId, tokenAddress, airSocialProfileExtraIds, profileExpiryTimestamp);
-    // create air social user registered transaction
-    createAirSocialUserRegisteredTransaction(
-      chainId,
-      airBlock,
-      transactionHash,
-      logOrCallIndex,
-      to,
-      airSocialUser.id,
-      airSocialProfile.id,
-      profileName,
-      from,
-      to,
-      tokenId,
-      tokenAddress,
-      airSocialUserExtraIds,
-      airSocialProfileExtraIds,
-      profileExpiryTimestamp,
-    );
-  }
+    /**
+     * @dev this function creates a AirSocialUserDefaultProfileChangeTransaction entity
+     * @param chainId chain id
+     * @param block air block entity
+     * @param transactionHash transaction hash
+     * @param logOrCallIndex log or call index
+     * @param from intiator of the transaction
+     * @param to protocol contract
+     * @param airSocialUser airSocialUser who's default profile changed
+     * @param newDefaultProfile new default AirSocialProfile
+     */
+    function createAirSocialUserDefaultProfileChangeTransaction(
```

Comment:
one more thing, can you please remove the spacing diff all over the file? its a lot of changes to look at.